### PR TITLE
Max/edit sites

### LINKED ISSUE #415: [Android] Edit Sites

### PR DESCRIPTION
**Pull request by maxwolffe**
Fixes #415

### CODE REVIEW

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Sites/SiteAbstractFragment.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        }
+        else {
+            EditSiteRequest editSiteRequest = new EditSiteRequest(mMainActivity, site, params, new Response.Listener<Site>() {
+                @Override
+                public void onResponse(Site site) {
+                    Log.e("successful site", "edit");
+                }
+            });
+            mNetworkManager.getRequestQueue().add(editSiteRequest);
+            mMainActivity.getSupportFragmentManager().popBackStack();
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Sites/SiteAbstractFragment.java]: why do we use back stack to go back here and use a new instance of the fragment above?

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Networking/Tasks/EditTaskRequest.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ protected static JSONObject taskParams(final Activity activity, final Task task)
 
         try {
             JSONObject taskJson = new JSONObject(mapper.writeValueAsString(task));
-            taskJson.put("due_date", new SimpleDateFormat("yyyy/MM/dd").format(task.getDueDate()));
+            taskJson.put("due_date", new SimpleDateFormat("yyyy/MM/dd").format(task.getDueDate())); //Should this be here?
```
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Networking/Tasks/EditTaskRequest.java]: should it?

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Sites/SiteAbstractFragment.java (hunk 1 above)
Comment (maxwolffe) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Sites/SiteAbstractFragment.java]: yeah, we can just pop in both cases.

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Networking/Tasks/EditTaskRequest.java (hunk 2 above)
Comment (maxwolffe) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Networking/Tasks/EditTaskRequest.java]: Nope

Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Networking/Tasks/EditTaskRequest.java]: Then we can probably just delete it.

File: watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Sites/SiteAbstractFragment.java (hunk 1 above)
Comment (nishdesai) [inline watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Sites/SiteAbstractFragment.java]: yeah let's do that

**Comment by maxwolffe** [comment]
don't merge yet, I'll pull in charles' changes.

**Comment by charlesx2013** [comment]
done! editing sites works now

**Comment by maxwolffe** [comment]
Cool. I think this is ready to go.

**Comment by maxwolffe** [comment]
ping @nishdesai

**Pull request merged**